conans/client/conan_api.py

```diff
@@ -681,6 +683,78 @@ def export_alias(self, reference, target_reference):
         target_reference = ConanFileReference.loads(str(target_reference))
         return self._manager.export_alias(reference, target_reference)
 
+    @api_method
+    def execute(self, command, references=None, profile_name=None,
+                initial_env=None, settings=None, options=None, remote=None,
+                build_modes=None, update=None, env=None, cwd=None):
+        """ Executes a command in the environment defined by the profile, the
+        specified references, settings, and options.
+
+        If any of the required packages is missing they are automatically
+        installed.
+        """
+        # setup proper defaults
+        cwd = cwd if cwd is not None else os.getcwd()
+        references = references if references is not None else []
+        new_env = initial_env if initial_env is not None else os.environ
+
+        # Load the profile given the specified settings, options, and env variables.
+        profile = profile_from_args(profile_name, settings, options, env, cwd,
+                                    self._client_cache)
+
+        # Build a temporary conanfile.txt, install it and determine the
+        # required environment variables. Note that, as long as extension
+        # is correct, the actual name does not matter.
+        env_vars = None
+
+        # Write all requested references to the file. Note that we have to
+        # close the file such that Windows can work with them.
+        # (see https://bugs.python.org/issue14243)
+        temp = tempfile.NamedTemporaryFile(suffix=".txt", delete=False)
```

Comment:
I'd suggest to just create the full file contents as a string in memory, then save it with the ``conans.files.save()`` method.

Reply:
I can do that but it would require changes to `conans.files.save()` given that the filename is generated temporary. Splitting the name generation into a call to `tempfile.mktemp` and `conans.files.save()` yields a potential race condition otherwise.

Reply:
the file is only needed for the self._manager.install, right?
may be it's better to provide an overload which accepts reference(s) instead of file name with reference(s)?

Reply:
I absolutely agree, I would also prefer to implement this `exec` functionality completely without writing any temporary file. However, I didn't wanted to do a potential large refactoring of the manager.install code (which I am not really familiar with) only for a prototype. Furthermore, given that it is not even sure yet that the `exec` feature is something that will be added to conan, increasing the size of the change set would certainly not improve my chances to get accepted. :smile: